addon/pattern-matchers.js

```diff
@@ -51,6 +89,74 @@ class SimplePatternMatchers {
     }
     return false;
   }
+
+  /**
+   * Generates a message indicating why no match was found.
+   * @param {Map<string, string>} testMap
+   * @returns {string}
+   */
+  generateReasonForNoMatching(testMap /* <Map<string, >> */) {
+    const paramNames = [...testMap.keys()].join(SEPARATOR);
+    let reason = `\nbut found no handler for it because\n`;
+
+    let paramNameMatches = [];
+    for (let paramMap of this.patternMapByQueryString.values()) {
+      if ([...paramMap.keys()].join(SEPARATOR) === paramNames) {
+        paramNameMatches.push(paramMap);
+      }
+    }
+
+    let requested = null;
+    let existing = null;
+
+    if (paramNameMatches.length > 0) {
+      requested = this.mapToText(testMap);
+      existing = [];
+      paramNameMatches.forEach((paramMap) =>
+        existing.push(this.mapToText(paramMap))
+      );
+
+      reason = `${reason}query values of:\n`;
+    } else {
+      requested = this.arryToText([...testMap.keys()]);
+      existing = [];
+      for (let paramMap of this.patternMapByQueryString.values()) {
+        existing.push(this.arryToText([...paramMap.keys()]));
+      }
+
+      reason = `${reason}query names of:\n`;
+    }
+
+    this._reasonForNoMatching =
+      `${reason}${requested}\n` +
+      `don't match any of:\n[\n\t${existing.join('\n\t')}\n]`;
+  }
+
+  /**
+   * Converts a Map object to a string in certain format.
+   * @param {Map<string, string>} the map object to be converted.
+   * @returns {string} the string converted to.
+   */
+  mapToText(map /* <string, string> */) {
+    let res =
+      (map &&
+        [...map.entries()].reduce((accu, entry) => {
+          accu.push(entry.join('='));
+          return accu;
+        }, [])) ||
```

Comment:
when we include the values, let's separate each param-value pair on its own line, so that  it'll be easier to see differences when you have very long values (very long names are much less likely IMHO)

```
don't match any of:
[
 {
 		foo=bar
 		baz=123
 },
 {
   foo=wut
   baz=okidoki
 }
]
```

WDYT?

Reply:
Fixed as suggested.